api/src/main/java/io/minio/MakeBucketArgs.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+public class MakeBucketArgs {
+  private final String bucketName;
+  private final String region;
+  private final boolean objectLock;
+  private static final String NULL_STRING = "(null)";
+
+  private MakeBucketArgs(Builder builder) {
+    this.bucketName = builder.bucketName;
+    this.region = builder.region;
+    this.objectLock = builder.objectLock;
+  }
+
+  public String bucketName() {
+    return bucketName;
+  }
+
+  public String region() {
+    return region;
+  }
+
+  public boolean objectLock() {
+    return objectLock;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static final class Builder {
+    private String bucketName;
+    private String region;
+    private boolean objectLock;
+
+    public Builder bucket(String bucketName) throws IllegalArgumentException {
+
+      if (bucketName == null) {
```

Comment:
1. `BucketName` and `ObjectName` are common type across `minio-py`. You could make these classes to ensure their correctness.
2. Any argument builder may accept `String bucketName` and/or `String objectName` and `BucketName` and `ObjectName` objects will be created internally.
3. My suggestion is to go through https://github.com/square/okhttp/blob/okhttp_3.14.x/okhttp/src/main/java/okhttp3/Headers.java as a reference for builder support.